### PR TITLE
UI: display CSR after generation

### PR DESCRIPTION
**Pull request by hellobontempo**
After generating an intermediate CSR this returns the data to the user so they can have it signed by the parent and then import the signed cert into the mount. If `type=exported` then the private key day is available to copy, `internal` or `existing` csr type displays `internal` (see screenshots below 📸 )
 Intermediate CSR generation happens in two places in the UI:

1. before the engine is configured (meaning, before issuers exist in the mount)
Screenshot 2023-02-09 at 5 20 54 PM

2. from the issuers index view
Screenshot 2023-02-09 at 4 36 16 PM

<hr>

### internal/existing

Screenshot 2023-02-09 at 2 57 55 PM

<hr> 

#### exported 

csr-exp

### CODE REVIEW

File: ui/app/models/pki/certificate/generate.js [hunk 1]
```diff
@@ -5,21 +5,20 @@ import PkiCertificateBaseModel from './base';
 
 const generateFromRole = [
   {
-    default: ['commonName'],
+    default: ['commonName', 'customTtl'],
```
Comment (hellobontempo) [inline ui/app/models/pki/certificate/generate.js:8]: leveraging joint component (also used in generating root)
Screenshot 2023-02-09 at 5 16 07 PM

File: ui/lib/pki/addon/components/pki-generate-csr.hbs [hunk 2]
```diff
@@ -1,28 +1,76 @@
...
+    <div class="field is-grouped is-fullwidth has-top-margin-l">
+      <div class="control">
+        <button type="button" class="button is-primary" {{on "click" @onComplete}} data-test-done>
+          Done
+        </button>
+      </div>
+    </div>
+  </footer>
+{{else}}
+  <form {{on "submit" (perform this.save)}}>
```
Comment (hellobontempo) [inline ui/lib/pki/addon/components/pki-generate-csr.hbs:48]: no changes here, just wrapped into an `{{else}}`

File: ui/lib/pki/addon/components/pki-generate-csr.hbs [hunk 3]
```diff
@@ -1,28 +1,76 @@
...
-      <button {{on "click" this.cancel}} type="button" class="button has-left-margin-s" data-test-cancel>
-        Cancel
-      </button>
-    </div>
-    {{#if this.alert}}
+    <div class="field is-grouped box is-fullwidth is-bottomless has-top-margin-l">
       <div class="control">
-        <AlertInline @type="danger" @paddingTop={{true}} @message={{this.alert}} @mimicRefresh={{true}} data-test-alert />
+        <button type="submit" class="button is-primary" data-test-save>
+          Generate
```
Comment (hellobontempo) [inline ui/lib/pki/addon/components/pki-generate-csr.hbs:63]: just kidding, changed this word

File: ui/lib/pki/addon/components/pki-sign-intermediate-form.hbs [hunk 4]
```diff
@@ -11,7 +11,7 @@
 
       {{#each this.showFields as |fieldName|}}
         {{#let (find-by "name" fieldName @model.allFields) as |attr|}}
-          <InfoTableRow @label={{or attr.options.label (humanize (dasherize attr.name))}} @value={{get @issuer attr.name}}>
+          <InfoTableRow @label={{or attr.options.label (humanize (dasherize attr.name))}} @value={{get @model attr.name}}>
```
Comment (hellobontempo) [inline ui/lib/pki/addon/components/pki-sign-intermediate-form.hbs:14]: came across this copy n paste typo

File: ui/lib/pki/addon/templates/keys/index.hbs [hunk 5]
```diff
@@ -8,7 +8,7 @@
   }}
   @isEngine={{true}}
 />
-{{#if this.model.hasConfig}}
+{{#if (or this.model.hasConfig this.model.keyModels)}}
```
Comment (hellobontempo) [inline ui/lib/pki/addon/templates/keys/index.hbs:11]: There's an intermediate state after CSR's are generated where keys exist, but no issuers - we want to list those keys in that case

**Review by zofskeez — APPROVED** [review]
:rocket:

**Pull request merged**